builtin/credential/aws/path_config_identity.go

```diff
@@ -0,0 +1,99 @@
+package awsauth
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/vault/helper/strutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathConfigIdentity(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "config/identity$",
+		Fields: map[string]*framework.FieldSchema{
+			"iam_alias": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Default:     identityAliasIAMUniqueID,
+				Description: fmt.Sprintf("Configure how the AWS auth method generates entity aliases when using IAM auth. Valid values are %q and %q", identityAliasIAMUniqueID, identityAliasIAMFullArn),
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation:   pathConfigIdentityRead,
+			logical.CreateOperation: pathConfigIdentityCreateUpdate,
+			logical.UpdateOperation: pathConfigIdentityCreateUpdate,
+		},
+
+		HelpSynopsis:    pathConfigIdentityHelpSyn,
+		HelpDescription: pathConfigIdentityHelpDesc,
+	}
+}
+
+func pathConfigIdentityRead(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	entry, err := req.Storage.Get(ctx, "config/identity")
+	if err != nil {
+		return nil, err
+	}
+	if entry == nil {
+		return nil, nil
+	}
+	var result identityConfig
+	if err := entry.DecodeJSON(&result); err != nil {
+		return nil, err
+	}
+	return &logical.Response{
+		Data: map[string]interface{}{
+			"iam_alias": result.IAMAlias,
+		},
+	}, nil
+}
+
+func pathConfigIdentityCreateUpdate(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	var configEntry identityConfig
+
+	iamAliasRaw, ok := data.GetOk("iam_alias")
+	if ok {
```

Comment:
The semantic of the "CreateUpdate" function works in a different way. First, it should depend on the `ExistenceCheck` to determine if the config entry exists. If the config entry doesn't exist, then the Operation will be set to CreateOperation, otherwise it will be set to UpdateOperation. Then, in the handler, we use the value of Operation to determine if the default value of the field should be used or not. (Please take a look at how it is used here: https://github.com/hashicorp/vault/blob/master/builtin/credential/approle/path_role.go#L926).

Reply:
Nope I don't think its important. Without the `ExistenceCheck`, the defined create operation func is a no-op. I mentioned it because your diff attempted to distinguish the operations here: https://github.com/hashicorp/vault/pull/5247/files#diff-99bcae2d8aefae0e6557d621c0027ac0R25. You can remove setting the `CreateOperation` there and rename the function as `pathConfigIdentityUpdate`.

Reply:
Ah, OK, makes sense now. Thanks.